### ISSUE TITLE
SSSD should do failover DNS lookups in parallel

### ISSUE THREAD

**Issue by pbrezina** [pbrezina/sssd-test#997]
Cloned from Pagure issue: https://pagure.io/SSSD/sssd/issue/993
- Created at 2011-09-12 15:44:34 by [sgallagh](https://pagure.io/user/sgallagh)
- Closed at 2020-03-24 14:22:08 as wontfix
- Assigned to nobody
---

DNS lookups while offline can take a very long time, especially with failover.

Right now, we behave as follows:

1) Start with the first failover server. Perform a DNS lookup against the first DNS server in the list. If that fails, try the seconds server, etc.

2) If nothing was resolved for the first server, try the next and repeat all of step 1.

Currently, we have a five second timeout to wait for responses from DNS servers. This means we are dealing with {{{5 * number_of_failover_servers * number_of_nameservers}}} seconds to time out if the network connection is unresponsive. This is FAR too long.

A better optimization would be to resolve all of the failover servers at the same time, rather than waiting for the earlier ones to fail. In this case, we would reduce this timeout to {{{5 * number_of_nameservers}}} (which may be further improved by ticket #976).

### Comments

---

**Comment from [sgallagh](https://pagure.io/user/sgallagh) at [2011-09-15 15:15:24](https://pagure.io/SSSD/sssd/issue/993#238303)**

Fields changed

milestone: NEEDS_TRIAGE => SSSD 1.8.0
priority: major => minor

---

**Comment from [sgallagh](https://pagure.io/user/sgallagh) at [2012-01-05 15:33:45](https://pagure.io/SSSD/sssd/issue/993#238304)**

The work done on ticket #976 mitigates the problem.

blockedby:  => 
blocking:  => 
milestone: SSSD 1.8.0 => SSSD Deferred

---

**Comment from [sgallagh](https://pagure.io/user/sgallagh) at [2012-01-19 03:22:28](https://pagure.io/SSSD/sssd/issue/993#238305)**

Fields changed

rhbz:  => 0

---

**Comment from [sgallagh](https://pagure.io/user/sgallagh) at [2017-02-24 14:57:57](https://pagure.io/SSSD/sssd/issue/993#238306)**

**Metadata Update from @sgallagh**:
- Issue assigned to jhrozek
- Issue set to the milestone: SSSD Patches welcome

---

**Comment from [sgallagh](https://pagure.io/user/sgallagh) at [2019-08-28 15:40:54](https://pagure.io/SSSD/sssd/issue/993#593232)**

**Metadata Update from @jhrozek**:
- Assignee reset

---

**Comment from [sgallagh](https://pagure.io/user/sgallagh) at [2020-03-24 14:22:07](https://pagure.io/SSSD/sssd/issue/993#637017)**

Thank you for taking time to submit this request for SSSD. Unfortunately this issue was not given priority and the team lacks the capacity to work on it at this time.

Given that we are unable to fulfill this request I am closing the issue as wontfix.

If the issue still persist on recent SSSD you can request re-consideration of this decision by reopening this issue. Please provide additional technical details about its importance to you.

Thank you for understanding.

---

**Comment from [sgallagh](https://pagure.io/user/sgallagh) at [2020-03-24 14:22:09](https://pagure.io/SSSD/sssd/issue/993#637018)**

**Metadata Update from @pbrezina**:
- Issue close_status updated to: wontfix
- Issue status updated to: Closed (was: Open)
